ddl/lightning/res_mgr.go

```diff
@@ -0,0 +1,368 @@
+	if err != nil {
+		return 0, errors.New(LERR_CREATE_ENGINE_FAILED)
+	}
+
+	if memRequire == firstAlloc {
+		m.structSize[ALLOC_ENGINE_INFO] = int64(unsafe.Sizeof(*ei))
+	}
+	// Count memory usage.
+	m.currUsage += m.structSize[ALLOC_ENGINE_INFO]
+	m.engineUsage += m.structSize[ALLOC_ENGINE_INFO]
+	log.L().Info(LINFO_OPEN_ENGINE, zap.String("backend key", bcKey),
+	zap.String("Engine key", engineKey),
+	zap.String("Current Memory Usage:", strconv.FormatInt(m.currUsage,10)),
+	zap.String("Memory limitation:", strconv.FormatInt(m.maxLimit, 10)))
+	return  workerCount, nil
+}
+
+// Create one
+func (m *LightningMemoryRoot) RegistWorkerContext(engineInfoKey string, id int) (*WorkerContext, error) {
+	var (
+		err           error = nil
+	    memRequire    int64 = 0
+		wCtx          *WorkerContext
+	)
+	m.mLock.Lock()
+	defer func() {
+		m.mLock.Unlock()
+	}()
+	// First to check the memory usage
+    m.totalMemoryConsume()
+	memRequire, err = m.checkMemoryUsage(ALLOC_WORKER_CONTEXT)
+	if err != nil {
+		log.L().Error(LERR_ALLOC_MEM_FAILED, zap.String("Engine key", engineInfoKey),
+		zap.String("worer Id:", strconv.Itoa(id)),
+		zap.String("Memory allocate:", strconv.FormatInt(memRequire, 10)),
+		zap.String("Current Memory Usage:", strconv.FormatInt(m.currUsage,10)),
+		zap.String("Memory limitation:", strconv.FormatInt(m.maxLimit, 10)))
+		return nil, err
+	}
+
+	wCtx = new(WorkerContext)
+	err = wCtx.InitWorkerContext(engineInfoKey, id)
+	if err != nil {
+		log.L().Error(LERR_CREATE_CONTEX_FAILED, zap.String("Engine key", engineInfoKey),
+		zap.String("worer Id:", strconv.Itoa(id)),
+		zap.String("Memory allocate:", strconv.FormatInt(memRequire, 10)),
+		zap.String("Current Memory Usage:", strconv.FormatInt(m.currUsage,10)),
+		zap.String("Memory limitation:", strconv.FormatInt(m.maxLimit, 10)))
+		return nil, err
+	}
+
+	if memRequire == firstAlloc {
+		m.structSize[ALLOC_ENGINE_INFO] = int64(unsafe.Sizeof(*wCtx))
+	}
+	// Count memory usage.
+	m.currUsage += m.structSize[ALLOC_WORKER_CONTEXT]
+	log.L().Info(LINFO_CREATE_WRITER, zap.String("Engine key", engineInfoKey),
+	zap.String("worer Id:", strconv.Itoa(id)),
+	zap.String("Memory allocate:", strconv.FormatInt(memRequire, 10)),
+	zap.String("Current Memory Usage:", strconv.FormatInt(m.currUsage,10)),
+	zap.String("Memory limitation:", strconv.FormatInt(m.maxLimit, 10)))
+	return wCtx, err
+}
+
+// Uniform entry to release KVCache slice related memory allocated
+func (m *LightningMemoryRoot) deleteBcEngine(bcKey string) error {
+	var err    error = nil
+	var count  int = 0
+	bc, exist := m.getBackendContext(bcKey)
+	if !exist {
+		log.L().Error(LERR_GET_BACKEND_FAILED, zap.String("backend key", bcKey))
+		return err
+	}
+	count = 0
+	// Delete EngienInfo registed in m.engineManager.engineCache
+	for _, ei := range bc.EngineCache {
+       eiKey := ei.key
+	   delete(m.EngineMgr.engineCache, eiKey)
+	   count++
+	}
+
+	bc.EngineCache = make(map[string]*engineInfo)
```

Comment:
maybe can use bc.EngineCache = make(map[string]*engineInfo), 
for range is not needed,
it will be gc by go.runtime

Reply:
or delete every map-kv ,should not renew the `EngineCache`